contracts/truefi2/TrueCreditAgency.sol

```diff
@@ -176,7 +188,7 @@ contract TrueCreditAgency is UpgradeableClaimable {
             bucket.cumulativeInterestPerShare = bucket.cumulativeInterestPerShare.add(
                 bucket.rate.mul(1e23).mul(timeNow.sub(bucket.timestamp)).div(365 days)
             );
-            bucket.rate = _creditScoreAdjustmentRate(uint8(i)).add(riskPremium);
+            bucket.rate = _currentRate(partialRate, _creditScoreAdjustmentRate(uint8(i)));
```

Comment:
Looks like we call `_currentRate()` with the same arguments as `currentRate()`. Can we deduplicate and just call `currentRate()` from here instead?

That should also let us change `currentRate()` to a one-liner:
min(riskPremium + utilizationAdjustment + creditScoreAdjustment, MAX_RATE_CAP).

Reply:
Thank you for pointing this out. Actually I was having doubts about the need of `_currentRate()` function. 

I really wanted to save some gas by utilizing precalculated sum of `riskPremium` and `utilizationAdjustmentRate()` stored as a local variable `partialRate`. Then my thoughts were that it would be nice to wrap adding and capping of `partialRate` and `creditScoreAdjustmentRate()` in a function which I named `_currentRate()`.

I agree this approach somewhat hurts readability. Maybe some comments and docstrings could improve it.

Reply:
I see your intent now.

If we haven't profiled and found a significant performance difference, I would prefer to keep the simpler implementation for code clarity. If you find a performance difference, document it so we know why we're opting for more convoluted code.

Reply:
Profiling showed gas use reduction by roughly 200_000 for 20 used buckets.

Reply:
Thanks for profiling this. That's a pretty nice improvement!